explorer.go

```diff
@@ -39,219 +49,333 @@ func (c *appContext) explorerUI(w http.ResponseWriter, r *http.Request) {
-
-	data := c.BlockData.GetBlockVerboseWithTxTypes(hash)
+func (exp *explorerUI) blockPage(w http.ResponseWriter, r *http.Request) {
+	hash := exp.app.getBlockHashCtx(r)
+	data := exp.app.BlockData.GetBlockVerboseWithStakeTxDetails(hash)
 	if data == nil {
 		apiLog.Errorf("Unable to get block %s", hash)
-		http.Error(w, http.StatusText(422), 422)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
-	str, err := TemplateExecToString(blockTemplate, "block", data)
+	str, err := TemplateExecToString(exp.templates[blockTemplateIndex], "block", data)
 	if err != nil {
-		http.Error(w, "template execute failure Error: "+err.Error(), http.StatusInternalServerError)
+		apiLog.Errorf("Template execute failure: %v", err)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
 	w.Header().Set("Content-Type", "text/html")
 	w.WriteHeader(http.StatusOK)
 	io.WriteString(w, str)
 }
 
-func (c *appContext) txPage(w http.ResponseWriter, r *http.Request) {
+func (exp *explorerUI) txPage(w http.ResponseWriter, r *http.Request) {
 	hash, ok := r.Context().Value(ctxTxHash).(string)
-
-	helpers := template.FuncMap{
-		"getTime":   getTime,
-		"getTotal":  getTotalapi,
-		"getAmount": getAmount,
-	}
-
-	txTemplate, _ := template.New("tx").Funcs(helpers).ParseFiles("views/tx.tmpl", "views/extras.tmpl")
-
 	if !ok {
 		apiLog.Trace("txid not set")
 		http.Error(w, "txid not set", http.StatusInternalServerError)
 		return
 	}
-	data := c.BlockData.GetRawTransaction(hash)
+	data := exp.app.BlockData.GetRawTransaction(hash)
 	if data == nil {
 		apiLog.Errorf("Unable to get transaction %s", hash)
-		http.Error(w, http.StatusText(422), 422)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
-	str, err := TemplateExecToString(txTemplate, "tx", data)
+	str, err := TemplateExecToString(exp.templates[txTemplateIndex], "tx", data)
 	if err != nil {
-		http.Error(w, "template execute failure Error: "+err.Error(), http.StatusInternalServerError)
+		apiLog.Errorf("Template execute failure: %v", err)
+		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
 	w.Header().Set("Content-Type", "text/html")
 	w.WriteHeader(http.StatusOK)
 	io.WriteString(w, str)
 }
 
-func (c *appContext) addressPage(w http.ResponseWriter, r *http.Request) {
+func (exp *explorerUI) addressPage(w http.ResponseWriter, r *http.Request) {
 	address, ok := r.Context().Value(ctxAddress).(string)
-
-	helpers := template.FuncMap{
-		"getTime":   getTime,
-		"getTotal":  getTotalapi,
-		"getAmount": getAmount,
-	}
-	txTemplate, _ := template.New("address").Funcs(helpers).ParseFiles("views/address.tmpl", "views/extras.tmpl")
-
 	if !ok {
 		apiLog.Trace("address not set")
 		http.Error(w, "address not set", http.StatusInternalServerError)
 		return
 	}
-	data := c.BlockData.GetAddressTransactions(address, 10)
+	data := exp.app.BlockData.GetAddressTransactions(address, 2000)
```

Comment:
I guess eventually this will be some kind of reactive display thing where you click for more, but it's probably OK like this for now.